app/authentication/no_questionnaire_state_exception.py

```diff
@@ -1,7 +1,10 @@
+from typing import Union
+
+
 class NoQuestionnaireStateException(Exception):
-    def __init__(self, value):
+    def __init__(self, value: Union[str, int]) -> None:
         super().__init__()
         self.value = value
 
-    def __str__(self):
-        return self.value
+    def __str__(self) -> str:
+        return self.value  # type: ignore
```

Comment:
We always raise this with an int (status code), but `str` comes from the fact that this inherits from base exception. The type for `__str__` just needs to be casted, then you don't need to ignore it.

```suggestion
    def __str__(self) -> str:
        return str(self.value)
```